### ISSUE TITLE
no-unused-var && no-redeclare: error when same variable name used in two for-loops in same scope

### ISSUE THREAD

**Issue by andrewdeandrade** [eslint/eslint#310]
I do not know if this is a bug or not, but the code block below makes these two rules throw. 

```
temp.js: line 3, col 9, Warning - i is already defined
temp.js: line 3, col 9, Warning - i is defined but never used
```

```
var arr1 = [1, 2];
var arr2 = [3, 4];
for (var i in arr1) {
    arr1[i] = 5;
}
for (var i in arr2) {
    arr2[i] = 10;
}
```

Default JSLint rules tell us to move the `var` statements to the beginning of the program/function. Default JSHint rules throw no errors for this code block. 

Furthermore, when I add the following code above as a test to `tests/lib/rules/no-unused-var.js`, it does not fail.

```
    "when evaluating 'var arr1 = [1, 2]; var arr = [3, 4]; for (var i in arr1) { arr1[i] = 5; } for (var i in arr2) { arr2[i] = 10; }": {
        topic: "var arr1 = [1, 2]; var arr2 = [3, 4]; for (var i in arr1) { arr1[i] = 5; } for (var i in arr2) { arr2[i] = 10; }",

        "should not report a violation": function(topic) {
            var config = { rules: {} };
            config.rules[RULE_ID] = 1;
            var messages = eslint.verify(topic, config);
            assert.equal(messages.length, 0);
        }
    }
```

Anyone have any idea on what the desired behavior here is? Is the scope of the for loop expression its own block or is the scope of that expression the function or program is it inside of?
